numpyro/infer/util.py

```diff
@@ -365,10 +365,10 @@ def initialize_model(rng_key, model,
                                                     model_kwargs=model_kwargs)
 
     init_strategy = init_strategy if isinstance(init_strategy, partial) else init_strategy()
-    if init_strategy.func is init_to_value:
+    if (init_strategy.func is init_to_value) and not replay_model:
```

Comment:
I added a check `not replay_model` in case there are dynamic supports in the model. If there is dynamic support, we will use the default `init_to_value` strategy (because given a constrained value, with dynamic support, the corresponding unconstrained value might change depending on the random seed). This is just a minor point.